src/test_report.py

```diff
@@ -96,32 +95,49 @@ def send_mail(self, email_msg, email_server):
         else:
             self._logger.log_message(logging.INFO, "Email Sent.")
 
+    def get_test_analysis(self, nodes):
+        total_runs = len(nodes)
+        total_failures = sum(node['status'] == "fail" for node in nodes)
+        total_timeouts = sum(node['status'] == "timeout" for node in nodes)
+        return total_runs, total_failures, total_timeouts
+
+    def create_test_report(self, root_node, child_nodes):
+        total_runs, total_failures, total_timeouts = \
+                    self.get_test_analysis(child_nodes)
```

Comment:
I think the PR should be rebased on top of https://github.com/kernelci/kernelci-pipeline/pull/67

Reply:
The issue is that #67 has many changes and dependencies and can't be easily merged now, which is why I created this PR to close #72 and get the pipeline back to a working state including email reports.  So I think the progression now should be:
* #90 
* #66 and remove
* #71 and remove
* Remove the `complete_hack` script
* #67 to improve email reports

I see now that some changes from #67 such as this one are included in the current PR #90:
https://github.com/kernelci/kernelci-pipeline/pull/67/commits/f2a65233c86a42ef6ef452c6361aaf7436380a98
So I should be able to cherry-pick that and rebase #90, then #67 could be rebased later.  I guess that's maybe what you meant, except we have to merge #90 before #66 to unblock things.

Reply:
If you want, I can drop commits related to `set_completed` from #67, and then we can get it merged first. Whatever you find suitable here.